web/packages/teleterm/src/ui/Search/pickers/ActionPicker.test.ts

```diff
@@ -0,0 +1,121 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { makeSuccessAttempt } from 'shared/hooks/useAsync';
+
+import { makeRootCluster } from 'teleterm/services/tshd/testHelpers';
+import { ResourceSearchError } from 'teleterm/ui/services/resources';
+
+import { getActionPickerStatus } from './ActionPicker';
+
+describe('getActionPickerStatus', () => {
+  it('partitions resource search errors into clusters with expired certs and non-retryable errors', () => {
+    const retryableError = new ResourceSearchError(
+      '/clusters/foo',
+      'server',
+      new Error('ssh: cert has expired')
+    );
+
+    const nonRetryableError = new ResourceSearchError(
+      '/clusters/bar',
+      'database',
+      new Error('whoops')
+    );
+
+    const status = getActionPickerStatus(
+      'foo',
+      [],
+      [],
+      [makeSuccessAttempt([])],
+      makeSuccessAttempt({
+        errors: [retryableError, nonRetryableError],
+        results: [],
+        search: 'foo',
+      })
```

Comment:
It would be easier to read if `getActionPickerStatus` used named arguments.

Reply:
Ah sorry, I was too lazy to type this all out.